src/main/java/com/google/ehub/servlets/UserRecommendationServlet.java

```diff
@@ -0,0 +1,35 @@
+package com.google.ehub.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.ehub.utility.UserRecommendationUtils;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Handles Get requests to get the recommended users for the user that is logged in.
+ */
+@WebServlet("/user-recommendation")
+public class UserRecommendationServlet extends HttpServlet {
```

Comment:
do you really need another servlet just for this? Can you just call your rec methods from the profile servlet?

Reply:
Oyin is returning user profile information in the ProfileServlet doGet() method. Do you think I should combine the data in that method into a single object and then send the user profile information and the recommendation list within the same JSON?

Reply:
OK I deleted this servlet and instead sent a new JSON object from ProfileServlet.

Reply:
I actually slightly preferred the UserRecommendationServlet since its separates the profile data from the recommendation engine. 

For example: If the recommendation algorithm gets more complex and takes longer to run, then the client can request the data from ProfileServlet separately and load that information first. Then the client can dynamically load the recommendation data. This way, the recommendation data doesn't block other parts of the user page from getting displayed. 

However, we are nearing the end of the internship so I think its reasonable to keep the single Servlet approach.

Reply:
I feel like there should be a separate servlet. I am looking at the changes that Bryan made and I think it will change the way I interact with the Profile Servlet.

Reply:
Don't worry Oyin, I made sure to change a small part of your code so that the profile page doesn't mess up. I already tested it and it's working the same way as before. I had it as a separate Servlet originally, but Rodrigo said that it would probably be better to combine it with the ProfileServlet to avoid making too many network calls.

Reply:
@BryanTriana , I really wish I knew about the changes ahead of time. I was under the impression that we were going to meet up so you could explain your modification plans and I could sign off on them. I didn't want a repeat of the time you changed the **Login Servlet** without telling me and it ending up braking my tests and calls.  But if you've tested everything with my original code and it's working, then it is fine by me. 👍

Reply:
@honey7100 Yeah I told you yesterday that I am going to have to modify some stuff in ProfileServlet. Today we can disscuss the changes in the debug meeting. Also when I modified the login servlet it didn't really break any tests because it was on the front end side so I wasn't able to detect the change. This time I made sure to run the website multiple times to check and see that nothing is missing.

Reply:
@BryanTriana we will talk about this in our meeting.